inc/saitam.h

```diff
@@ -0,0 +1,898 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc
+ *
+ * @file    saitam.h
+ *
+ * @brief   This module defines SAI Bridge
+ */
+
+#if !defined (__SAITAM_H_)
+#define __SAITAM_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAITAM SAI - Telemetry and monitoring specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief TAM Statistic ID
+ *
+ * Identifies a specific counter within the SAI object hierarchy
+ */
+typedef enum _sai_tam_stat_attr_t
+{
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_STAT_ATTR_START,
+
+    /**
+     * @brief Monitored object id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_BUFFER_POOL, SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP
+     */
+    SAI_TAM_STAT_ATTR_PARENT_ID = SAI_TAM_STAT_ATTR_START,
+
+    /**
+     * @brief Monitored object type
+     *
+     * Eg.: SAI_OBJECT_TYPE_BUFFER_POOL, SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP
+     * Not used for input. Optional for output.
+     *
+     * @type sai_object_type_t
```

Comment:
I agree wit kcudnik, I suggest to remove it for this version. If later we really need this for performance reason, let's them add it back. I am not really convinced that we need this for now.